### PR TITLE
Add doc for zero-length rule

### PR DESCRIPTION
**Pull request by jeddy3**
I quite like writing these docs as it helps me familiarize myself with the expected behaviour of each rule :)

### CODE REVIEW

File: src/rules/number-zero-length-no-unit/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+Disallow units for lengths of zero.
+
+```css
+    a { top: 0px; }
+/**          ↑ ↑
+ * This zero and this type of length unit */
+```
+
+For zero lengths the unit identifier is optional.
```
Comment (davidtheclark) [inline src/rules/number-zero-length-no-unit/README.md]: We should probably have a brief explanation of how a "length" differs from other value types, why `0%` and `0s` are accepted. We might even list out the units that will be checked?

Comment (MoOx) [inline src/rules/number-zero-length-no-unit/README.md]: I need that explanation indeed. I don't know why 0% and 0s are accepted.

**Comment by jeddy3** [comment]
I've expanded the description to:

a. explain what a length is.
b. list the applicable units. 

The description is butchered from here: http://www.w3.org/TR/css3-values/#lengths.

**Comment by davidtheclark** [comment]
Nice.

**Comment by jeddy3** [comment]
Shall I merge?

**Pull request merged**